rdr_service/dao/bq_questionnaire_dao.py

```diff
@@ -113,9 +113,20 @@ def make_bqrecord(self, p_id, module_id, latest=False, convert_to_enum=False):
             responses = session.execute(_participant_module_responses_sql, {'module_id': module_id, 'p_id': p_id,
                                                                             'test_hpo': TEST_HPO_NAME,
                                                                             'system': PPI_SYSTEM})
+            answer_hashes_processed = []
             for qr in responses:
+                if qr.answer_hash in answer_hashes_processed:
+                    # PDR-863:  This response is pending being flagged by RDR as a duplicate (same answer hash as
+                    # a more recently created response).  Proactively exclude from PDR data to avoid orphaned records
+                    continue
+
+                answer_hashes_processed.append(qr.answer_hash)
```

Comment:
Does this only compare answer hashes? We could receive some responses multiple times that have the same answer hash (such as responding multiple times to EHR or GROR consent). Using external id as part of the comparison could help make sure distinct responses that have the same answers are accounted for.